src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -2084,27 +2082,36 @@ public function registerNodeTypes($types, $allowUpdate)
                 $query = "SELECT * FROM phpcr_type_nodes WHERE name = ?";
                 $result = $this->getConnection()->fetchColumn($query, array($type->getName()));
                 if ($result) {
-                    $this->getConnection()->delete('phpcr_type_nodes', array('node_type_id' => $result));
                     $this->getConnection()->delete('phpcr_type_props', array('node_type_id' => $result));
                     $this->getConnection()->delete('phpcr_type_childs', array('node_type_id' => $result));
```

Comment:
so we delete all props and children mappings for the type, but not the type entry itself. this could lead us to an inconsistent situation when updating fails for some reason. but i don't see a better solution